src/discovery/mdns.js

```diff
@@ -0,0 +1,119 @@
+import { TypedEmitter } from 'tiny-typed-emitter'
+import net from 'node:net'
+import NoiseSecretStream from '@hyperswarm/secret-stream'
+import { once } from 'node:events'
+import dnssd from 'dnssd'
+import b4a from 'b4a'
+
+
+const SERVICE_NAME = 'mapeo'
+
+
+/** @typedef {{ publicKey: Buffer, secretKey: Buffer }} Keypair */
+
+/**
+ * @typedef {Object} Service
+ * @property {string} host - hostname of the service
+ * @property {number} port - port of the service
+ * @property {Object<string, any>} txt - TXT records of the service
+ * @property {string[]} [addresses] - addresses of the service
+ */
+
+
+export class MdnsDiscovery extends TypedEmitter {
+  #identityKeypair
+  #server
+  /** @type {Set<string>} */
+  #connections = new Set()
```

Comment:
```suggestion
  #socketConnections = new Set()
  #noiseConnections = new Set()
```
We're actually doing two stages of dedupe - not sure the second is strictly necessary, but good to have just in case.